developer/src/kmcmplib/src/meson.build

```
@@ -28,7 +28,7 @@ if cpp_compiler.get_id() == 'emscripten'
   # wasm-exceptions supported in Node 18+, Chrome 95+, Firefox 100+, Safari 15.2+
   flags += ['-fwasm-exceptions']
   lib_links = ['--whole-archive', '-sMODULARIZE', '-sEXPORT_ES6']
-  links += ['-fwasm-exceptions', '--bind', '-sEXPORTED_RUNTIME_METHODS=[\'UTF8ToString\']']
+  links += ['-fwasm-exceptions', '--bind', '-sEXPORTED_RUNTIME_METHODS=[\'UTF8ToString\',\'wasmExports\']']
```

Doesn't this ensure `this.Module.wasmExports` is always available? Do we still need to fallback to Module.asm?

---

`wasmExports` is only available on 3.1.44.  Hooray for a smooth transition! 😆 

on 3.1.43 we get:

> warning: invalid item in EXPORTED_RUNTIME_METHODS: wasmExports

There may be another way to do this as `asm` is apparently internal, but I'm not sure what it is.

---

... if we update the prereq version of emcripten from 3.1.32 to 3.1.44

https://github.com/keymanapp/keyman/blob/master/docs/build/windows.md#web-dependencies

---

@darcywong00 as written here it works either way.

---

> There may be another way to do this as `asm` is apparently internal, but I'm not sure what it is.

We'd need to conditionalize it on the emscripten version which is nicely self-documenting in the meson.build. Something like:

```suggestion
  links += ['-fwasm-exceptions', '--bind']
  if cpp_compiler.version() >= '3.1.44'
    # emscripten 3.1.44 removes .asm object and so we need to export `wasmExports`
    # #9375; https://github.com/emscripten-core/emscripten/blob/main/ChangeLog.md#3144---072523
    links += ['-sEXPORTED_RUNTIME_METHODS=[\'UTF8ToString\',\'wasmExports\']']
  else
    # emscripten < 3.1.44 does not include `wasmExports`
    links += ['-sEXPORTED_RUNTIME_METHODS=[\'UTF8ToString\']']
  endif
```

---

why? the wasmExports message is only a warning.

---

My goal is zero messages in our builds (it's a stretch goal for sure). Every extraneous warning is more noise to sift through for actual real warnings which need to be addressed.